### ISSUE TITLE
histPlot() Error: Breaks and labels are different lengths

### ISSUE THREAD

**Issue by jooyoungseo** [massimoaria/bibliometrix#82]
When using `histPlot()` function, it throws an error as follows:

> Error: Breaks and labels are different lengths

Would you mind addressing this issue?

Below is the reproducible code:

``` r
bib <- "https://jooyoungseo.github.io/test.bib"

# Loading required packages:
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(bibliometrix))

# Importing metadata from BibTeX entries:
D <- bibliometrix::readFiles(bib)

# Export the metadata into a data_frame excluding some early access articles:
M <- bibliometrix::convert2df(D, dbsource = "wos", format = "bibtex") %>%
  dplyr::filter(DT != "ARTICLE, EARLY ACCESS")
#> 
#> Converting your wos collection into a bibliographic dataframe
#> 
#> Articles extracted   100 
#> Articles extracted   200 
#> Articles extracted   300 
#> Articles extracted   400 
#> Articles extracted   413 
#> Done!
#> 
#> 
#> Generating affiliation field tag AU_UN from C1:  Done!

# Create a historical citation network
options(width=130)
histResults <- histNetwork(M, min.citations = 10, sep = ";")
#> Articles analysed   100 
#> Articles analysed   200 
#> Articles analysed   252

# Plot a historical co-citation network
histPlot(histResults, n=15, size = 10, labelsize=5, verbose = FALSE)
#> Warning: The curvature argument has been deprecated in favour of strength
#> Error: Breaks and labels are different lengths
```

![](https://i.imgur.com/JIUgc5E.png)

<sup>Created on 2019-11-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

<details style="margin-bottom:10px;">

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> - Session info -----------------------------------------------------------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       America/New_York            
#>  date     2019-11-13                  
#> 
#> - Packages ---------------------------------------------------------------------------------------------------------------------
#>  package         * version     date       lib source                                    
#>  assertthat        0.2.1       2019-03-21 [1] CRAN (R 3.6.1)                            
#>  backports         1.1.5       2019-10-02 [1] CRAN (R 3.6.1)                            
#>  bibliometrix    * 2.3.1       2019-11-12 [1] Github (massimoaria/bibliometrix@7b2f764) 
#>  broom             0.5.2.9002  2019-10-14 [1] Github (tidymodels/broom@01181bb)         
#>  cellranger        1.1.0       2016-07-27 [1] CRAN (R 3.6.1)                            
#>  cli               1.1.0       2019-03-19 [1] CRAN (R 3.6.1)                            
#>  cluster           2.1.0       2019-06-19 [1] CRAN (R 3.6.1)                            
#>  colorspace        1.4-1       2019-03-18 [1] CRAN (R 3.6.1)                            
#>  crayon            1.3.4       2017-09-16 [1] CRAN (R 3.6.1)                            
#>  curl              4.2         2019-09-24 [1] CRAN (R 3.6.1)                            
#>  data.table        1.12.7      2019-11-08 [1] Github (Rdatatable/data.table@412a781)    
#>  DBI               1.0.0       2018-05-02 [1] CRAN (R 3.6.0)                            
#>  dbplyr            1.4.2       2019-11-08 [1] Github (hadley/dbplyr@99d4db4)            
#>  digest            0.6.22      2019-10-21 [1] CRAN (R 3.6.1)                            
#>  dplyr           * 0.8.3.9000  2019-11-08 [1] Github (hadley/dplyr@b357f06)             
#>  DT                0.9         2019-09-17 [1] CRAN (R 3.6.1)                            
#>  evaluate          0.14        2019-05-28 [1] CRAN (R 3.6.1)                            
#>  factoextra        1.0.5       2017-08-22 [1] CRAN (R 3.6.0)                            
#>  FactoMineR        1.42        2019-07-03 [1] CRAN (R 3.6.1)                            
#>  farver            1.1.0       2018-11-20 [1] CRAN (R 3.6.0)                            
#>  fastmap           1.0.1       2019-10-08 [1] CRAN (R 3.6.1)                            
#>  flashClust        1.01-2      2012-08-21 [1] CRAN (R 3.6.0)                            
#>  forcats         * 0.4.0       2019-02-17 [1] CRAN (R 3.6.1)                            
#>  foreign           0.8-72      2019-08-02 [1] CRAN (R 3.6.1)                            
#>  fs                1.3.1       2019-05-06 [1] CRAN (R 3.6.1)                            
#>  generics          0.0.2       2018-11-29 [1] CRAN (R 3.6.1)                            
#>  ggforce           0.3.1       2019-08-20 [1] CRAN (R 3.6.1)                            
#>  ggplot2         * 3.2.1.9000  2019-11-08 [1] Github (hadley/ggplot2@e58be03)           
#>  ggraph            2.0.0.9000  2019-10-03 [1] Github (thomasp85/ggraph@0e86312)         
#>  ggrepel           0.8.1       2019-05-07 [1] CRAN (R 3.6.1)                            
#>  glue              1.3.1.9000  2019-11-08 [1] Github (tidyverse/glue@f8dc26f)           
#>  graphlayouts      0.5.0.9000  2019-09-02 [1] Github (schochastics/graphlayouts@6a21d81)
#>  gridExtra         2.3         2017-09-09 [1] CRAN (R 3.6.0)                            
#>  gtable            0.3.0       2019-03-25 [1] CRAN (R 3.6.1)                            
#>  haven             2.2.0.9000  2019-11-08 [1] Github (tidyverse/haven@690f9d7)          
#>  highr             0.8         2019-03-20 [1] CRAN (R 3.6.1)                            
#>  hms               0.5.2       2019-10-30 [1] CRAN (R 3.6.1)                            
#>  htmltools         0.4.0       2019-10-04 [1] CRAN (R 3.6.1)                            
#>  htmlwidgets       1.5.1.9000  2019-11-08 [1] Github (ramnathv/htmlwidgets@b357323)     
#>  httpuv            1.5.2       2019-09-11 [1] CRAN (R 3.6.1)                            
#>  httr              1.4.1       2019-08-05 [1] CRAN (R 3.6.1)                            
#>  igraph            1.2.4.1     2019-04-22 [1] CRAN (R 3.6.1)                            
#>  jsonlite          1.6         2018-12-07 [1] CRAN (R 3.6.1)                            
#>  knitr             1.26.1      2019-11-13 [1] Github (yihui/knitr@bf98f2a)              
#>  labeling          0.3         2014-08-23 [1] CRAN (R 3.6.0)                            
#>  later             1.0.0       2019-10-04 [1] CRAN (R 3.6.1)                            
#>  lattice           0.20-38     2018-11-04 [1] CRAN (R 3.6.1)                            
#>  leaps             3.0         2017-01-10 [1] CRAN (R 3.6.0)                            
#>  lifecycle         0.1.0       2019-08-01 [1] CRAN (R 3.6.1)                            
#>  lubridate         1.7.4.9000  2019-11-08 [1] Github (hadley/lubridate@a6ee2f7)         
#>  magrittr          1.5         2014-11-22 [1] CRAN (R 3.6.1)                            
#>  MASS              7.3-51.4    2019-03-31 [1] CRAN (R 3.6.1)                            
#>  Matrix            1.2-17      2019-03-22 [1] CRAN (R 3.6.1)                            
#>  mime              0.7         2019-06-11 [1] CRAN (R 3.6.0)                            
#>  modelr            0.1.5       2019-08-08 [1] CRAN (R 3.6.1)                            
#>  munsell           0.5.0       2018-06-12 [1] CRAN (R 3.6.1)                            
#>  networkD3         0.4         2017-03-18 [1] CRAN (R 3.6.0)                            
#>  openxlsx          4.1.3       2019-11-07 [1] CRAN (R 3.6.1)                            
#>  pillar            1.4.2       2019-06-29 [1] CRAN (R 3.6.1)                            
#>  pkgconfig         2.0.3       2019-09-22 [1] CRAN (R 3.6.1)                            
#>  plyr              1.8.4       2016-06-08 [1] CRAN (R 3.6.1)                            
#>  polyclip          1.10-0      2019-03-14 [1] CRAN (R 3.6.0)                            
#>  promises          1.1.0       2019-10-04 [1] CRAN (R 3.6.1)                            
#>  purrr           * 0.3.3.9000  2019-11-12 [1] Github (hadley/purrr@0168a59)             
#>  R6                2.4.1       2019-11-12 [1] CRAN (R 3.6.1)                            
#>  RColorBrewer      1.1-2       2014-12-07 [1] CRAN (R 3.6.0)                            
#>  Rcpp              1.0.3.1     2019-11-13 [1] Github (RcppCore/Rcpp@8580d28)            
#>  readr           * 1.3.1.9000  2019-11-07 [1] Github (hadley/readr@7fccc36)             
#>  readxl            1.3.1.9000  2019-11-07 [1] Github (hadley/readxl@43cf0eb)            
#>  reprex            0.3.0.9001  2019-11-13 [1] Github (tidyverse/reprex@27aa69a)         
#>  reshape2          1.4.3       2017-12-11 [1] CRAN (R 3.6.1)                            
#>  rio               0.5.16      2018-11-26 [1] CRAN (R 3.6.0)                            
#>  RISmed            2.1.7       2017-06-06 [1] CRAN (R 3.6.0)                            
#>  rlang             0.4.1.9000  2019-11-12 [1] Github (r-lib/rlang@5a0b80a)              
#>  rmarkdown         1.16.4      2019-11-11 [1] Github (rstudio/rmarkdown@1f63943)        
#>  rscopus           0.6.6       2019-09-17 [1] CRAN (R 3.6.1)                            
#>  rvest             0.3.5.9000  2019-11-09 [1] Github (hadley/rvest@a5200f2)             
#>  scales            1.0.0       2018-08-09 [1] CRAN (R 3.6.1)                            
#>  scatterplot3d     0.3-41      2018-03-14 [1] CRAN (R 3.6.0)                            
#>  sessioninfo       1.1.1       2018-11-05 [1] CRAN (R 3.6.1)                            
#>  shiny             1.4.0       2019-10-10 [1] CRAN (R 3.6.1)                            
#>  shinycssloaders   0.2.0       2017-05-12 [1] CRAN (R 3.6.0)                            
#>  shinythemes       1.1.2       2018-11-06 [1] CRAN (R 3.6.0)                            
#>  SnowballC         0.6.0       2019-01-15 [1] CRAN (R 3.6.0)                            
#>  stringdist        0.9.5.5     2019-10-21 [1] CRAN (R 3.6.1)                            
#>  stringi           1.4.3       2019-03-12 [1] CRAN (R 3.6.0)                            
#>  stringr         * 1.4.0.9000  2019-11-11 [1] Github (hadley/stringr@80aaaac)           
#>  styler            1.2.0.9000  2019-11-13 [1] Github (r-lib/styler@a8acde5)             
#>  tibble          * 2.1.3       2019-06-06 [1] CRAN (R 3.6.1)                            
#>  tidygraph         1.1.2.9999  2019-10-03 [1] Github (thomasp85/tidygraph@d0cdea0)      
#>  tidyr           * 1.0.0.9000  2019-11-14 [1] Github (hadley/tidyr@cb51247)             
#>  tidyselect        0.2.99.9000 2019-11-13 [1] Github (tidyverse/tidyselect@d5b1609)     
#>  tidyverse       * 1.2.1.9000  2019-11-09 [1] Github (tidyverse/tidyverse@b122d97)      
#>  tweenr            1.0.1       2018-12-14 [1] CRAN (R 3.6.0)                            
#>  vctrs             0.2.0.9007  2019-11-13 [1] Github (r-lib/vctrs@390bc5a)              
#>  viridis           0.5.1       2018-03-29 [1] CRAN (R 3.6.0)                            
#>  viridisLite       0.3.0       2018-02-01 [1] CRAN (R 3.6.1)                            
#>  withr             2.1.2       2018-03-15 [1] CRAN (R 3.6.1)                            
#>  xfun              0.11.1      2019-11-12 [1] Github (yihui/xfun@ef9908c)               
#>  xml2              1.2.2.9000  2019-11-06 [1] Github (r-lib/xml2@a0b698e)               
#>  xtable            1.8-4       2019-04-21 [1] CRAN (R 3.6.0)                            
#>  yaml              2.2.0       2018-07-25 [1] CRAN (R 3.6.0)                            
#>  zeallot           0.1.0       2018-01-28 [1] CRAN (R 3.6.1)                            
#>  zip               2.0.4       2019-09-01 [1] CRAN (R 3.6.1)                            
#> 
#> [1] C:/Program Files/R/R-3.6.1/library
```

</details>

**Comment by massimoaria** [comment]
I've replicated your code and everything works fine

> bib <- "https://jooyoungseo.github.io/test.bib"
> 
> # Loading required packages:
> suppressPackageStartupMessages(library(tidyverse))
Warning messages:
1: package ‘tidyverse’ was built under R version 3.6.1 
2: package ‘tibble’ was built under R version 3.6.1 
3: package ‘dplyr’ was built under R version 3.6.1 
> suppressPackageStartupMessages(library(bibliometrix))
> 
> # Importing metadata from BibTeX entries:
> D <- bibliometrix::readFiles(bib)
> 
> # Export the metadata into a data_frame excluding some early access articles:
> M <- bibliometrix::convert2df(D, dbsource = "wos", format = "bibtex") %>%
+     dplyr::filter(DT != "ARTICLE, EARLY ACCESS")

Converting your wos collection into a bibliographic dataframe

Articles extracted   100 
Articles extracted   200 
Articles extracted   300 
Articles extracted   400 
Articles extracted   413 
Done!


Generating affiliation field tag AU_UN from C1:  Done!

> options(width=130)
> histResults <- histNetwork(M, min.citations = 10, sep = ";")
Articles analysed   100 
Articles analysed   200 
Articles analysed   252 
> #> Articles analysed   100 
> #> Articles analysed   200 
> #> Articles analysed   252
> 
> # Plot a historical co-citation network
> histPlot(histResults, n=15, size = 10, labelsize=5, verbose = FALSE)
IGRAPH a01c33b DN-- 4 2 -- 
+ attr: name (v/c), id (v/c), size (v/n), years (v/n), color (e/c)
+ edges from a01c33b (vertex names):
[1] SCHWARZ BB, 2011, J LEARN SCI->CHIN C, 2010, J LEARN SCI   SANDOVAL W, 2014, J LEARN SCI->DERRY SJ, 2010, J LEARN SCI

![image](https://user-images.githubusercontent.com/16023543/68848261-4f341900-06d0-11ea-9918-60298f32a15d.png)

**Comment by jooyoungseo** [comment]
This issue has been resolved with #81. I am closing this. Thank you very much!
